Source/WebCore/html/HTMLElement.h

```diff
@@ -90,7 +90,12 @@ class HTMLElement : public StyledElement {
 
     TextDirection computeDirectionality() const;
     bool hasDirectionAuto() const;
-    TextDirection directionalityIfhasDirAutoAttribute(bool& isAuto) const;
+
+    struct TextDirectionAutoResult {
+        TextDirection direction;
+        bool isAuto { false };
+    };
+    TextDirectionAutoResult directionalityAndIsAuto() const;
```

Comment:
I find this function difficult to understand. Despite its name it does not simply return directionality and also a boolean to indicate whether auto applies; the auto state affects the directionality. Separately, I think `isAuto` is subtle and ambiguous; what is auto and what does auto mean? I think it might need a different name and return type to clarify what’s going on. The old name seems slightly clearer. Maybe just revert to that name and return an `optional`? We would have to use `value_or` a little bit but I think it might be worth it for slightly better clarity.

Reply:
I'd rename it to directionalityIfDirIsAuto and make it return std::optional<TextDirection>.